Snippets/CosmosDB/CosmosDB_1/Transactions.cs

```diff
@@ -0,0 +1,182 @@
+    }
+
+    void ExtractPartitionKeyFromHeadersCustom(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractPartitionKeyFromHeadersCustom
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractPartitionKeyFromHeaders(new CustomPartitionKeyFromHeadersExtractor());
+
+        #endregion
+    }
+
+    void ExtractContainerInfoFromHeader(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractContainerInfoFromHeader
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromHeader("ContainerKey", headerValue => new ContainerInformation(headerValue, new PartitionKeyPath("/partitionKey")));
+
+        #endregion
+    }
+
+    void ExtractContainerInfoFromHeaders(PersistenceExtensions<CosmosPersistence> persistence, MyAppsCustomConfigurationHolder appConfig)
+    {
+        #region ExtractContainerInfoFromHeaders
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromHeaders<MyAppsCustomConfigurationHolder>((headers, config) => new ContainerInformation(headers["ContainerNameHeader"], new PartitionKeyPath(appConfig.PartitionKeyPath)), appConfig);
+
+        #endregion
+    }
+
+    void ExtractContainerInfoFromHeadersCustom(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractContainerInfoFromHeadersCustom
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromHeaders(new CustomContainerInformationFromHeadersExtractor());
+
+        #endregion
+    }
+
+    void ExtractPartitionKeyFromMessageExtractor(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractPartitionKeyFromMessageExtractor
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractPartitionKeyFromMessage<MyMessage>(message => new PartitionKey(message.ItemId));
+
+        #endregion
+    }
+
+    void ExtractPartitionKeyFromMessageCustom(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractPartitionKeyFromMessageCustom
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractPartitionKeyFromMessage<MyMessage>(message => new PartitionKey(message.ItemId));
+
+        #endregion
+    }
+
+
+    void ExtractContainerInfoFromMessageExtractor(PersistenceExtensions<CosmosPersistence> persistence)
+    {
+        #region ExtractContainerInfoFromMessageExtractor
+
+        var transactionInformation = persistence.TransactionInformation();
+        transactionInformation.ExtractContainerInformationFromMessage<MyMessage>(message => new ContainerInformation(message.ItemId.ToString(), new PartitionKeyPath("/partitionKey")));
```

Comment:
I think it would be more realistic to show here a hard-coded container name instead of a super dynamic one, no?

Reply:
agreed